### ISSUE TITLE
Incorrect catch-all behaviour 

### ISSUE THREAD

**Issue by remkop22** [ibraheemdev/matchit#30]
The catch-all path `/*p` currently doesn't match `/`, Although the documentation states the opposite. Copying the code for catch-alls in the README, the first assertions panics.
```rust
let mut m = Router::new();
m.insert("/*p", true)?;

assert_eq!(m.at("/")?.params.get("p"), Some(""));
assert_eq!(m.at("/foo.js")?.params.get("p"), Some("foo.js"));
assert_eq!(m.at("/c/bar.css")?.params.get("p"), Some("c/bar.css"));
```

**Comment by remkop22** [comment]
Reading the change-log I understand this behavior was consciously changed when going from `0.5.0` to `0.6.0`, so it seems like it's just outdated docs. I'll be using `0.5.0` for now instead.

**Comment by ibraheemdev** [comment]
Yes, the behavior was changed to be more flexible, you can register the route twice if you want to old behavior. The docs should be updated.

**Comment by ibraheemdev** [comment]
The docs have been updated.
